rowhammer_tester/scripts/rowhammer.py

```diff
@@ -364,17 +378,48 @@ def rand_row():
             'rand_per_row': patterns_random_per_row,
         }[args.pattern]
 
-        count = args.read_count if not args.read_count_range else args.read_count_range[0]
+        if args.read_count_range:
+            count = args.read_count_range[0]
+        elif args.read_count:
+            count = args.read_count
         count_stop = count if not args.read_count_range else args.read_count_range[1]
         count_step = 1 if not args.read_count_range else args.read_count_range[2]
 
+        row_pair_distance = 2
         while count <= count_stop:
-            row_hammer.run(row_pairs=row_pairs, read_count=count, pattern_generator=pattern)
+            row_hammer.err_summary["{}".format(count)] = {"read_count": count}
+            if args.all_rows:
+                for i in range(args.start_row, args.nrows - row_pair_distance, args.row_jump):
+                    row_pairs = [(0 + i, 0 + i + row_pair_distance)]
+                    err_in_rows = row_hammer.run(
+                        row_pairs=row_pairs, read_count=count, pattern_generator=pattern)
+                    row_hammer.err_summary["{}".format(count)]["pair_{}_{}".format(
+                        row_pairs[0][0], row_pairs[0][1])] = {
+                            "hammer_row_1": row_pairs[0][0],
+                            "hammer_row_2": row_pairs[0][1],
+                            "errors_in_rows": err_in_rows
+                        }
+            else:
+                err_in_rows = row_hammer.run(
+                    row_pairs=row_pairs, read_count=count, pattern_generator=pattern)
+                if args.row_pairs == 'const':
+                    row_hammer.err_summary["{}".format(count)]["pair_{}_{}".format(
+                        row_pairs[0][0], row_pairs[0][1])] = {
+                            "hammer_row_1": row_pairs[0][0],
+                            "hammer_row_2": row_pairs[0][1],
+                            "errors_in_rows": err_in_rows
+                        }
+                else:
+                    row_hammer.err_summary["{}".format(count)]["sequential_attacks"] = {
+                        "row_pairs": row_pairs,
+                        "errors_in_rows": err_in_rows
+                    }
+
             count += count_step
             if row_hammer.bitflip_found and args.exit_on_bit_flip:
                 break
 
-    if len(row_hammer.err_summary):
+    if row_hammer.log_directory:
```

Comment:
Why did you have to change the condition here?

Reply:
I had to change it because earlier, when no bitflips appeared, `row_hammer.err_summary` was empty and file wasn't written. Now the data is written even if no bitflips appear, so can keep track on what configuration don't cause bitflips.
When `args.log_dir` is not specified, it evaluates to False, and not output file is produced - it was changed in the PR for cell retention measurments. Should we manage it differently?